### PR TITLE
EE-7789: Oauth Token Expiration - refresh dance sends redirect_uri pa…

### PR DESCRIPTION
**Pull request by sofiamorseletto**
…rameter in the query

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/AuthorizationCodeOAuthHandler.java [hunk 1]
```diff
@@ -197,6 +205,11 @@ private AuthorizationCodeOAuthDancer createDancer(AuthorizationCodeConfig config
         .customParameters(config.getCustomParameters())
         .customParametersExtractorsExprs(getParameterExtractors(config));
 
+    if (!featureFlaggingService.isEnabled(NO_OAUTH_REDIRECT_URI)) {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/AuthorizationCodeOAuthHandler.java:208]: I don't think the fix should be applied here but in the AuthorizationCodeDancer in the oauth-service. By applying it here, the issue still persists if somebody tries to consume the Dropbox API directly using the Http Connector. The AuthorizationCodeDancerBuilder should expose a method that allows us to configure this behavior. This class should merely use the feature flag to determine how to configure that. At the same time, the dancer's default behavior should also depend on the feature flag

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/AuthorizationCodeOAuthHandler.java (hunk 1 above)
Comment (sofiamorseletto) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/AuthorizationCodeOAuthHandler.java:208]: Hi, Mariano! A change has also been made on the `DefaultAuthorizationCodeOAuthDancer` of the Oauth-client, you can check it on this PR: https://github.com/mulesoft/mule-oauth-client/pull/69 . If no additional parameters are added the client does not send the redirect_uri parameter when refreshing the token. So even if somebody tries to consume the Dropbox API using the HTTP Connector, the parameters wouldn't have been added. Does this PR make sense to you considering also the change on https://github.com/mulesoft/mule-oauth-client/pull/69 ?

Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/AuthorizationCodeOAuthHandler.java:208]: Hello Sofia. I didn't get the notification for that other PR, thanks for sending it over. I added some additional comments on that one too. I still don't think this is the right solution. Actually, now that I think about it I see additional problems with this solution:

* The expectation when using a cloud connector is that it works out of the box. This shouldn't depend on a feature flag. The only reason the feature flag is actually needed is the solution not being backwards compatible. It should be up to each connector to specify whether the redirect_uri should be provided or not. The runtime cannot take that decision on its own.
* This solution is that it wouldn't work if the Dropbox connector were to be used alongside another one which does need the redirect_uri parameter, as the feature flag would affect one of the connectors all the time.

**Review by marianogonzalez — CHANGES_REQUESTED** [review]
The jira issue should be moved to the MULE project as the fix is going on the open source modules and the EE project is not public

**Pull request closed without merging**